openedx_tagging/core/tagging/rest_api/v1/serializers.py

```diff
@@ -192,3 +192,67 @@ def get_children_count(self, obj):
         Returns the number of child tags of the given tag.
         """
         return len(obj.sub_tags)
+
+
+class TaxonomyTagCreateBodySerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+    Serializer of the body for the Taxonomy Tags CREATE view
+    """
+
+    tag = serializers.CharField(required=True)
+    parent_tag_value = serializers.CharField(
+        source='parent.value', required=False
+    )
+    external_id = serializers.CharField(required=False)
+
+    def validate_parent_tag_value(self, value):
+        """
+        Check that the provided parent Tag exists based on the value
+        """
+        valid = Tag.objects.filter(value__iexact=value).exists()
+        if not valid:
+            raise serializers.ValidationError("Invalid `parent_tag_value` provided")
```

Comment:
Personally I would put this validation in the `Taxonomy.add_tag` method, where most other validation is anyways. That way it gets used for both the python API and REST API. Currently, only the REST API has this validation. You can leave it as-is though.

Removing the validation method(s) from these three serializers would also make the serializer code much more compact.

Reply:
After trying it out, it's definitely more concise and having the validations all in one place is cleaner, thanks @pomegranited and @bradenmacdonald for the suggestions! I updated it.